diff --git a/app/config/windowsettings.cpp b/app/config/windowsettings.cpp
index 6e8a68f0479b5ce7599756b5088e4540a32372cf..4a87a522706c79b8d8ba0d67173bc34c22424c27 100644
--- a/app/config/windowsettings.cpp
+++ b/app/config/windowsettings.cpp
@@ -10,15 +10,15 @@
 #include <KLocalizedString>
 #include <KMessageBox>
 
-WindowSettings::WindowSettings(QWidget *parent)
+WindowSettings::WindowSettings(QWidget *parent, const QList<QString> &outputOrder)
     : QWidget(parent)
 {
     setupUi(this);
 
-    for (int i = 2; i <= QGuiApplication::screens().count(); i++)
-        kcfg_Screen->insertItem(i, xi18nc("@item:inlistbox", "Screen %1", i));
+    for (int i = 1; i <= outputOrder.count(); i++)
+        kcfg_Screen->insertItem(i, xi18nc("@item:inlistbox", "Screen %1", i) + QStringLiteral(" (%1)").arg(outputOrder[i - 1]));
 
-    if (QGuiApplication::screens().count() > 1) {
+    if (outputOrder.count() > 1) {
         screenLabel->setEnabled(true);
         kcfg_Screen->setEnabled(true);
     }
diff --git a/app/config/windowsettings.h b/app/config/windowsettings.h
index 7fa28df318912a061579bfd654bad76e6e29d757..e6b38a89d2bfb3eb8c71888a2c5c842ca3a70fa5 100644
--- a/app/config/windowsettings.h
+++ b/app/config/windowsettings.h
@@ -14,7 +14,7 @@ class WindowSettings : public QWidget, private Ui::WindowSettings
     Q_OBJECT
 
 public:
-    explicit WindowSettings(QWidget *parent = nullptr);
+    explicit WindowSettings(QWidget *parent, const QList<QString> &outputOrder);
     ~WindowSettings() override;
 
 Q_SIGNALS:
diff --git a/app/config/windowsettings.ui b/app/config/windowsettings.ui
index 58470a3163c11d7217c90475dcf9268541c97884..0adf6776275c217eebe54edbe554823bdf178765 100644
--- a/app/config/windowsettings.ui
+++ b/app/config/windowsettings.ui
@@ -348,11 +348,6 @@
             <string comment="@item:inlistbox">At mouse location</string>
            </property>
           </item>
-          <item>
-           <property name="text">
-            <string comment="@item:inlistbox">Screen 1</string>
-           </property>
-          </item>
          </widget>
         </item>
         <item>
diff --git a/app/mainwindow.cpp b/app/mainwindow.cpp
index 3708874e4dfad152c829592380e6c25315b70897..e1ebc8120bb11a9ab5246ab10de630fcebf28c33 100644
--- a/app/mainwindow.cpp
+++ b/app/mainwindow.cpp
@@ -761,7 +761,7 @@ void MainWindow::updateScreenMenu()
     action->setChecked(Settings::screen() == 0);
 
     for (int i = 1; i <= m_outputOrderWatcher->outputOrder().count(); i++) {
-        action = m_screenMenu->addAction(xi18nc("@item:inmenu", "Screen %1", i));
+        action = m_screenMenu->addAction(xi18nc("@item:inmenu", "Screen %1", i) + QStringLiteral(" (%1)").arg(m_outputOrderWatcher->outputOrder()[i - 1]));
         action->setCheckable(true);
         action->setData(i);
         action->setChecked(i == Settings::screen());
@@ -883,7 +883,7 @@ void MainWindow::configureApp()
     settingsDialog->setFaceType(KPageDialog::List);
     connect(settingsDialog, &KConfigDialog::settingsChanged, this, &MainWindow::applySettings);
 
-    WindowSettings *windowSettings = new WindowSettings(settingsDialog);
+    WindowSettings *windowSettings = new WindowSettings(settingsDialog, m_outputOrderWatcher->outputOrder());
     settingsDialog->addPage(windowSettings, xi18nc("@title Preferences page name", "Window"), QStringLiteral("preferences-system-windows-move"));
     connect(windowSettings, &WindowSettings::updateWindowGeometry, this, &MainWindow::setWindowGeometry);
 
