diff --git a/PKGBUILD b/PKGBUILD
index e37d5e4..cbfb48b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -42,14 +42,25 @@ makedepends=(extra-cmake-modules
              plasma-wayland-protocols)
 groups=(kde-applications
         kde-utilities)
-source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pkgver.tar.xz{,.sig})
+source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pkgver.tar.xz{,.sig}
+        geometry.patch
+        cleanup.patch
+        names.patch)
 sha256sums=('ed5676a65ed99e57b7bfd73fbd31f93e7f5a8d4dc54551d3c2a6668f78b5b5b2'
-            'SKIP')
+            'SKIP'
+            '8ca673801b9def6b77d4c11df52894512cfd3289cd069c88b70b4deeee4a6201'
+            '540851c603b861ebbde4078eea5b9aae4da9ea35ff7ad235d9820e4c1c029a58'
+            'c6c99ceca8ba880578b5e1d742caa96dfc9339a4c5c7b1c17165a8067a803875')
 validpgpkeys=(CA262C6C83DE4D2FB28A332A3A6A4DB839EAA6D7  # Albert Astals Cid <aacid@kde.org>
               F23275E4BF10AFC1DF6914A6DBD2CE893E2D1C87  # Christoph Feck <cfeck@kde.org>
               D81C0CB38EB725EF6691C385BB463350D6EF31EF) # Heiko Becker <heiko.becker@kde.org>
 
 build() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i ../geometry.patch
+  patch -Np1 -i ../cleanup.patch
+  patch -Np1 -i ../names.patch
+  cd ..
   cmake -B build -S $pkgname-$pkgver \
     -DBUILD_TESTING=OFF
   cmake --build build
