diff --git a/PKGBUILD b/PKGBUILD
index cd97470..ace2621 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,11 +10,14 @@ license=('MIT')
 replaces=('betterdiscord')
 makedepends=('git' 'nodejs' 'yarn')
 depends=('libxss')
-source=("$pkgname-${pkgver}.tar.gz::$url/archive/v${pkgver}.tar.gz")
-sha256sums=('aecc83656770ff4d91aa0c4650e4774c162d47766e149ab2b8e0287ec52c4857')
+source=("$pkgname-${pkgver}.tar.gz::$url/archive/v${pkgver}.tar.gz"
+        "paths.patch")
+sha256sums=('aecc83656770ff4d91aa0c4650e4774c162d47766e149ab2b8e0287ec52c4857'
+            "92fadb54c7540c5a70306cf9e1b53418d61588e0516a756d1052b9c1d7335ab9")
 
 build() {
     cd Installer-${pkgver}
+    patch -Np1 -i ../paths.patch
     yarn && yarn dist:dir
 }
 
